### DISCUSSION TITLE
form表单和field字段的交互模式，readOnly模式下应不可操作

### DISCUSSION THREAD

**Discussion by li9269391** [alibaba/formily#1589] [Q&A 2.x] [answered]
form 表单和 field 字段交互模式：
`
type FormPatternTypes = 'editable' | 'disabled' | 'readOnly' | 'readPretty'；

interface setPattern {
  (pattern?: FormPatternTypes): void
}
`
如设置 Field 或 Form  的交互模式为 readOnly，发现仅 Input 输入框为可读，但 Select 和 checkbox 还可以操作选择、切换
例子：
https://codesandbox.io/s/ancient-platform-6eh2n?file=/App.tsx

**Comment by janryWang** [comment, marked as answer]
![image](https://user-images.githubusercontent.com/4060976/122217941-78d2a900-cee0-11eb-938f-b25ebe58af85.png)
看样子是Antd自身的问题，可以给Antd提个issue

**Reply by li9269391** [reply]
好的，明白了

**Comment by liuweiGL** [comment]
`field` 字段只是一个 model（纯粹管理状态的），对应的 ui 行为你需要看你使用的是哪个 ui 库，然后去查看 ui 库的文档。
你现在用的是 antd，那你去看看它的 select 组件支持  readOnly 吗？

**Reply by li9269391** [reply]
谢谢，确实是 antd 的 select 组件不支持 readOnly

**Comment by John251314** [comment]
哎，这个事情着实让人迷惑，虽然意识到了应该是组件的问题

**Comment by John251314** [comment]
readPretty可以
